packages/ui-components/src/ApplePay/utilities.ts

```diff
@@ -59,26 +88,142 @@ export function buildSession(
     shippingType: "shipping",
   };
 
+  const paymentOverrides = config.paymentOverrides || {};
+
   const request = new PaymentRequest(
-    paymentMethodData,
-    paymentDetails,
+    paymentOverrides.paymentMethodData
+      ? paymentOverrides.paymentMethodData
+      : paymentMethodData,
+    paymentOverrides.paymentDetails
+      ? paymentOverrides.paymentDetails
+      : paymentDetails,
```

Comment:
It could be possible to leverage the `??` operator here if it makes sense?
```suggestion
    paymentOverrides.paymentMethodData ?? paymentMethodData,
    paymentOverrides.paymentDetails ?? paymentDetails,
```

Reply:
Yeah good shout. I went with `||` however to catch all all falsey values